core/primitives/src/types.rs

```diff
@@ -152,6 +152,9 @@ pub enum StateChangeCause {
     /// State change that happens when we update validator accounts. Not associated with with any
     /// specific transaction or receipt.
     ValidatorAccountsUpdate,
+    /// State change that is happens due to migration that happens in first block of an epoch
```

Comment:
nit:

```suggestion
    /// State change that happens due to state-correction migration that happens in the first block of an epoch
```

Ideally, reference this PR as an example for the reader, so they don't need to do git-blame to look up what kind of "state-corrections" happen.